sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_client_manager.py

```diff
@@ -264,15 +268,119 @@ def __exit__(self, *args):
+        self._original_connection_string = connection_string
+        self._credential = credential
+        self._user_agent = user_agent
+        self._retry_total = retry_total
+        self._retry_backoff_max = retry_backoff_max
+        self._replica_discovery_enabled = replica_discovery_enabled
+        self._next_update_time = time.time() + MINIMAL_CLIENT_REFRESH_INTERVAL
+        self._args = dict(kwargs)
         self._min_backoff_sec = min_backoff_sec
         self._max_backoff_sec = max_backoff_sec
 
-    def set_clients(self, replica_clients: List[ConfigurationClientWrapper]):
-        self._replica_clients.clear()
-        self._replica_clients.extend(replica_clients)
+        failover_endpoints = find_auto_failover_endpoints(endpoint, replica_discovery_enabled)
+        if connection_string and endpoint:
+            self._replica_clients.append(
+                ConfigurationClientWrapper.from_connection_string(
+                    endpoint, connection_string, user_agent, retry_total, retry_backoff_max, **self._args
+                )
+            )
+            for failover_endpoint in failover_endpoints:
+                failover_connection_string = connection_string.replace(endpoint, failover_endpoint)
+                self._replica_clients.append(
+                    ConfigurationClientWrapper.from_connection_string(
+                        failover_endpoint,
+                        failover_connection_string,
+                        user_agent,
+                        retry_total,
+                        retry_backoff_max,
+                        **self._args
+                    )
+                )
+            return
+        if endpoint and credential:
+            self._replica_clients.append(
+                ConfigurationClientWrapper.from_credential(
+                    endpoint, credential, user_agent, retry_total, retry_backoff_max, **self._args
+                )
+            )
+            for failover_endpoint in failover_endpoints:
+                self._replica_clients.append(
+                    ConfigurationClientWrapper.from_credential(
+                        failover_endpoint, credential, user_agent, retry_total, retry_backoff_max, **self._args
+                    )
+                )
+            return
+        raise ValueError("Please pass either endpoint and credential, or a connection string with a value.")
+
+    def refresh_clients(self):
+        if not self._replica_discovery_enabled:
+            return
+        if self._next_update_time > time.time():
+            return
+        failover_endpoints = find_auto_failover_endpoints(self._original_endpoint, self._replica_discovery_enabled)
+
+        if failover_endpoints is None:
+            self._next_update_time = time.time() + FALLBACK_CLIENT_REFRESH_EXPIRED_INTERVAL
+            return
+
+        if len(failover_endpoints) == 0:
+            self._next_update_time = time.time() + MINIMAL_CLIENT_REFRESH_INTERVAL
+            return
```

Comment:
What is the difference between these two scenarios?

Reply:
So, I wrote this weird because None means something went wrong. I guess we could also have it throw an exception and catch it right away, whereas [] is just that we found the SRV record, it just didn't have any replicas.

Reply:
Updated with a comment for now at least.